src/version_file.rs

```diff
@@ -49,20 +56,17 @@ impl<'a> VersionFile<'a> {
 }
 
 #[cfg(not(test))]
-#[cfg(any(target_os = "macos", target_os = "linux"))]
-fn home_path() -> PathBuf {
-    PathBuf::from(env!("HOME"))
-}
-
-#[cfg(not(test))]
-#[cfg(target_os = "windows")]
-fn home_path() -> PathBuf {
-    PathBuf::from(env!("USERPROFILE")) // Or use $HOMEPATH?
+fn cache_path() -> Result<PathBuf, Error> {
+    let project_dir = directories::ProjectDirs::from("", "", "update-informer-rs")
+        .map_or(Err("Unable to find cache directory"), Ok)?;
+    let directory = project_dir.cache_dir().to_path_buf();
+    fs::create_dir_all(&directory)?;
```

Comment:
Can we avoid creating directories?

Reply:
It seems like the standards for macOS and Windows really do want you to have directories.